src/lib/InstantSearch.js

```diff
@@ -312,6 +312,10 @@ See: https://www.algolia.com/doc/guides/building-search-ui/going-further/backend
    * @return {undefined} This method does not return anything
    */
   dispose() {
+    this.scheduleSearch.cancel();
+    this.scheduleRender.cancel();
+    clearTimeout(this._searchStalledTimer);
```

Comment:
We could use the defer approach for the stalled timer, the implementation would be the exact same one. But right now, the `defer` function does not accept a timeout since it relies on `Promise`. It's easy to change its implementation to use `setTimeout` but will have to update the tests. For the moment this implementation is good enough.